File: shared/src/types/game.ts
import type { CardString } from './card.js';
import type { Street } from './hand.js';
import type { PublicPlayerState } from './player.js';

export type GameType = 'NLHE' | 'PLO';

export interface GameConfig {
  gameType: GameType;
  smallBlind: number;
  bigBlind: number;
  maxBuyIn: number;
  actionTimeSeconds: number;
  minPlayers: number;
  maxPlayers: number;
}

export type GamePhase =
  | 'waiting_for_players'
  | 'hand_in_progress'
  | 'hand_complete'
  | 'paused';

export interface PotDisplay {
  amount: number;
  eligible: string[];
}

export interface GameState {
  phase: GamePhase;
  config: GameConfig;
  handNumber: number;
  players: PublicPlayerState[];
  communityCards: CardString[];
  secondBoard?: CardString[];
  pots: PotDisplay[];
  currentStreet: Street | null;
  dealerSeatIndex: number;
  currentActorSeatIndex: number | null;
  actionTimeRemaining: number;
}

File: ios/PokerSofta/Sources/Models/Game.swift
import Foundation

enum GameType: String, Codable {
    case NLHE, PLO
}

enum GamePhase: String, Codable {
    case waitingForPlayers = "waiting_for_players"
    case handInProgress = "hand_in_progress"
    case handComplete = "hand_complete"
    case paused
}

struct GameConfig: Codable {
    let gameType: GameType
    let smallBlind: Double
    let bigBlind: Double
    let maxBuyIn: Double
    let actionTimeSeconds: Int
    let minPlayers: Int
    let maxPlayers: Int
}

struct PotDisplay: Codable {
    let amount: Double
    let eligible: [String]
}

struct GameState: Codable {
    let phase: GamePhase
    let config: GameConfig
    let handNumber: Int
    let players: [PublicPlayerState]
    let communityCards: [CardString]
    let secondBoard: [CardString]?
    let pots: [PotDisplay]
    let currentStreet: Street?
    let dealerSeatIndex: Int
    let currentActorSeatIndex: Int?
    let actionTimeRemaining: Int
}
